### ISSUE TITLE
test_expanduser fails on master/windows on github CI

### ISSUE THREAD

**Issue by yarikoptic** [datalad/datalad#5338]
Just spotted within datalad/git-annex sweep of tests (ok on released, maint):
Tests were passing 5 days back, so possibly triggered by #5310

https://github.com/datalad/git-annex/runs/1722165033?check_suite_focus=true

```
======================================================================
ERROR: datalad.core.distributed.tests.test_clone.test_expanduser
----------------------------------------------------------------------
Traceback (most recent call last):
  File "C:\hostedtoolcache\windows\Python\3.7.9\x64\lib\site-packages\nose\case.py", line 198, in runTest
    self.test(*self.arg)
  File "C:\hostedtoolcache\windows\Python\3.7.9\x64\lib\site-packages\datalad\tests\utils.py", line 757, in _wrap_with_tempfile
    return t(*(arg + (filename,)), **kw)
  File "C:\hostedtoolcache\windows\Python\3.7.9\x64\lib\site-packages\datalad\tests\utils.py", line 757, in _wrap_with_tempfile
    return t(*(arg + (filename,)), **kw)
  File "C:\hostedtoolcache\windows\Python\3.7.9\x64\lib\site-packages\datalad\core\distributed\tests\test_clone.py", line 545, in test_expanduser
    cloneds = clone(op.join('~', 'src'), 'dest')
  File "C:\hostedtoolcache\windows\Python\3.7.9\x64\lib\site-packages\datalad\interface\utils.py", line 482, in eval_func
    return return_func(generator_func)(*args, **kwargs)
  File "C:\hostedtoolcache\windows\Python\3.7.9\x64\lib\site-packages\datalad\interface\utils.py", line 470, in return_func
    results = list(results)
  File "C:\hostedtoolcache\windows\Python\3.7.9\x64\lib\site-packages\datalad\interface\utils.py", line 457, in generator_func
    msg="Command did not complete successfully")
datalad.support.exceptions.IncompleteResultsError: Command did not complete successfully. 1 failed:

[{'action': 'install',
  'message': ('Failed to clone from any candidate source URL. Encountered '
              'errors per each url were:\n'
              '- %s',
              '~\\src\n'
              '  CommandError: \'"git" "clone" "--progress" '
              '"C:\\Users\\runneradmin\\AppData\\Local\\Temp\\datalad_temp_zl9rmwun\\src" '
              '"C:\\Users\\runneradmin\\AppData\\Local\\Temp\\datalad_temp_4aq19uux\\dest"\' '
              "failed with exitcode 128 [err: 'Cloning into "
              "'C:\\Users\\runneradmin\\AppData\\Local\\Temp\\datalad_temp_4aq19uux\\dest'...\n"
              'fatal: '
              "'C:\\Users\\runneradmin\\AppData\\Local\\Temp\\datalad_temp_zl9rmwun\\src' "
              'does not appear to be a git repository\n'
              'fatal: Could not read from remote repository.\n'
              '\n'
              'Please make sure you have the correct access rights\n'
              "and the repository exists.']"),
  'path': 'C:\\Users\\runneradmin\\AppData\\Local\\Temp\\datalad_temp_4aq19uux\\dest',
  'source_url': '~\\src',
  'status': 'error',
  'type': 'dataset'}]
```

**Comment by mih** [comment]
Works fine on appveyor, I am not able to replicate on a windows box either.

https://ci.appveyor.com/project/mih/datalad/builds/37297560/job/7th4w1w81okevvuv#L575

**Comment by yarikoptic** [comment]
I think it is Python 3.7(.9) specific. Passes for me in a conda env with python 3.8.5, but fails with 3.7.9.

**Comment by yarikoptic** [comment]
Gy, with 3.8 it fails on maint and release. So those changes (whatever they were, didn't analyze yet) really catered 3.8 ;-) https://github.com/datalad/git-annex/pull/45/checks?check_run_id=1729910975

**Comment by yarikoptic** [comment]
Yep, almost certain https://github.com/datalad/datalad/pull/5310/commits/4cbfe579d342937fd227106d39cecb02a18015ac
and due to https://bugs.python.org/issue36264 . Interesting to check how it behaves now under msys2

**Comment by yarikoptic** [comment]
ok, fixed in master so closing - will be included in 0.14.0
